### PR TITLE
KAFKA-6581: Fix the ConsumerGroupCommand indefinite execution if one of the partition is unavailable.

### PR DESCRIPTION
**Pull request by SahilAggarwal**
* Checks if partition available before calling consumer.position
* Adds timeout on consumer.position() call.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by SahilAggarwal** [comment]
@ijuma 0.10.0 branch doesn't seem to have jenkins.sh due to which above checks are failing. Can you please suggest what need to be done next?

File: core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala [hunk 1]
```diff
@@ -332,12 +348,33 @@ object ConsumerGroupCommand {
     }
 
     protected def getLogEndOffset(topic: String, partition: Int): LogEndOffsetResult = {
-      val consumer = getConsumer()
-      val topicPartition = new TopicPartition(topic, partition)
-      consumer.assign(List(topicPartition).asJava)
-      consumer.seekToEnd(List(topicPartition).asJava)
-      val logEndOffset = consumer.position(topicPartition)
-      LogEndOffsetResult.LogEndOffset(logEndOffset)
+      zkUtils.getLeaderForPartition(topic, partition) match {
```
Comment (hachikuji) [inline core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala:351]: A better approach would be to use `KafkaConsumer.endOffsets`. Although its doc claims that it blocks indefinitely, it is in fact bound by the request timeout. I will submit a separate patch to correct this.

Comment (SahilAggarwal) [reply core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala:351]: KafkaConsumer in 0.10.0 doesn't have endOffsets. It was introduced from 0.10.1 i guess. 0.10.0 doesn't have request timeout too.

Comment (hachikuji) [reply core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala:351]: Hmm, that's true. Is there any particular reason you need this in 0.10.0? We are unlikely to do another bug fix release for that version. Note that it should be possible to use tools from more recent versions on an 0.10.0 broker.

Comment (SahilAggarwal) [reply core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala:351]: We rely on this for consumer offset metrics and lose the visibility even if single partition becomes unavailable. We could but didn't want to maintain 2 versions just for this.
I see 0.10.0.2 as unreleased, no possibility there?

Comment (hachikuji) [reply core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala:351]: It's very unlikely we'll release 0.10.0.2 at this point. Your next closest version is 0.11.0.2, which already has the fix that I suggested above.

**Pull request closed without merging**